Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -452,10 +467,37 @@ WarpX::computePhiCartesian (const amrex::Vector<std::unique_ptr<amrex::MultiFab>
     mlmg.setVerbose(verbosity);
     mlmg.setMaxIter(max_iters);
     mlmg.solve( GetVecOfPtrs(phi), GetVecOfConstPtrs(rho), required_precision, 0.0);
+
+#ifdef AMREX_USE_EB
+    // use amrex to directly calculate the electric field since with EB's the
+    // simple finite difference scheme in WarpX::computeE sometimes fails
+    if (do_electrostatic == ElectrostaticSolverAlgo::LabFrame)
+    {
+        for (int lev = 0; lev <= max_level; ++lev) {
+#if (AMREX_SPACEDIM==2)
+            mlmg.getGradSolution(
+                {amrex::Array<amrex::MultiFab*,2>{
+                    get_pointer_Efield_fp(lev, 0),get_pointer_Efield_fp(lev, 2)
+                    }}
+            );
+#elif (AMREX_SPACEDIM==3)
+            mlmg.getGradSolution(
+                {amrex::Array<amrex::MultiFab*,3>{
+                    get_pointer_Efield_fp(lev, 0),get_pointer_Efield_fp(lev, 1),
+                    get_pointer_Efield_fp(lev, 2)
+                    }}
+            );
+#endif
+            get_pointer_Efield_fp(lev, 0)->mult(-1._rt);
+            get_pointer_Efield_fp(lev, 1)->mult(-1._rt);
```

Comment:
Do we need to multiply `Ey` by `-1` in 2D?

Reply:
Nope, that is definitely just a performance hit. I've now fixed it. Thanks for catching it!